File: backend/semgrep_rules/tests/java/kisa_sw_37_missing_error_handling.java
class kisa_sw_37_missing_error_handling {
    void unsafe() {
        try {
            risky();
        } catch (Exception e) {
            // ruleid: kisa.sw37.java.empty-catch-block
        }
    }
    void safe() {
        try {
            risky();
        } catch (Exception e) {
            // ok: kisa.sw37.java.empty-catch-block
            throw new RuntimeException(e);
        }
    }
    void risky() {}
}

File: backend/semgrep_rules/tests/javascript/kisa_sw_37_missing_error_handling.js
function unsafe() {
  try {
    risky();
  } catch (e) {
    // ruleid: kisa.sw37.javascript.empty-catch-block
  }
}
function safe() {
  try {
    risky();
  } catch (e) {
    // ok: kisa.sw37.javascript.empty-catch-block
    throw e;
  }
}
function risky() {}
